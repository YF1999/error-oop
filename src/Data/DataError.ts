import { IErrorArguments, IErrorOptions, NativeError } from '../Native';
import { ErrorTool } from '../Tools';

export interface IDataErrorOptions extends IErrorOptions {}

/**
 * Applicable when an error occurs on or with an external data source.
 */
export class DataError extends NativeError {
    public constructor();
    /**
     * @param message The error message that explains the reason for this error.
     */
    public constructor(message: string);
    /**
     * @param message The error message that explains the reason for this error.
     * @param innerError The error that is the cause of the current error. Stack trace will be appended.
     */
    public constructor(message: string, innerError: Error);
    /**
     * @param options The constructor options.
     */
    public constructor(options: IDataErrorOptions);

    public constructor(...args: IErrorArguments<IDataErrorOptions>) {
        super(ErrorTool.parseErrorArguments([], ...args));
    }
}
