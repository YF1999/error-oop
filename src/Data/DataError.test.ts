import { DataError } from './DataError';

standardTestError(DataError, 'DataError', 'error');
