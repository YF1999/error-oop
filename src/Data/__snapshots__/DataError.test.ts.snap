// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test \`DataError\` own keys 1`] = `
Array [
  "stack",
]
`;

exports[`Test \`DataError\` own keys 2`] = `
Array [
  "stack",
  "message",
]
`;

exports[`Test \`DataError\` own keys 3`] = `
Array [
  "stack",
  "message",
]
`;

exports[`Test \`DataError\` own keys 4`] = `
Array [
  "stack",
  "message",
]
`;
